### PR TITLE
Add docs for tuist task plugins

### PR DESCRIPTION
**Pull request by fortmarek**
### Short description 📝

This PR adds documentation for how to use the new tasks plugins. 

Once this is merged, we should be close to merging the new plugins architecture to `main` 🎉 

### How to test the changes locally 🧐

Read the docs 🙂 

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: projects/docs/docs/commands/plugin.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+slug: '/commands/plugin'
+title: Interact with plugins
+description: 'Learn how to use plugin commands for developing and publishing tuist plugins.'
+---
```
Comment (danyf90) [inline projects/docs/docs/commands/plugin.md:5]: This is just for Tasks, right? I would specify it doesn't apply to project description helpers

**Review by danyf90 — APPROVED** [review]

File: projects/docs/docs/commands/plugin.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+slug: '/commands/plugin'
+title: Interact with plugins
+description: 'Learn how to use plugin commands for developing and publishing tuist plugins.'
```
Comment (pepicrft) [inline projects/docs/docs/commands/plugin.md:4]: Nitpick: `tuist` -> `Tuist`

File: projects/docs/docs/commands/plugin.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+---
+slug: '/commands/plugin'
+title: Interact with plugins
```
Comment (pepicrft) [inline projects/docs/docs/commands/plugin.md:3]: What do you think about `Sharing via plugins` instead of `Interact with plugins`?

**Review by pepicrft — APPROVED** [review]
Brilliant work @fortmarek

File: projects/docs/docs/commands/plugin.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+### Test
+
+`test` subcommand is used for testing the plugin.
+
+#### Arguments
+
+| Argument | Short | Description | Default | Required |
+| ------------------ | ----- | ------------------------- | ------- | -------- |
+| `--configuration` | `-c`  | Choose configuration | `debug` | No |
+| `--path`  | `-p`  | Path to the directory that contains the definition of the plugin |  | No |
```
Comment (ezraberch) [inline projects/docs/docs/commands/plugin.md:34]: To be consistent with how it's done elsewhere (such as [here](https://docs.tuist.io/commands/generate))

```suggestion
| `--path`  | `-p`  | Path to the directory that contains the definition of the plugin | Current directory | No |
```

File: projects/docs/docs/commands/plugin.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+### Archive
+
+Archives a plugin into a `NameOfPlugin.tuist-plugin.zip`. This file is then used for publishing the plugin as described [here](/plugins/creating-plugins).
+
+#### Arguments
+
+| Argument | Short | Description | Default | Required |
+| ------------------ | ----- | ------------------------- | ------- | -------- |
+| `--path`  | `-p`  | Path to the directory that contains the definition of the plugin |  | No |
```
Comment (ezraberch) [inline projects/docs/docs/commands/plugin.md:46]:
```suggestion
| `--path`  | `-p`  | Path to the directory that contains the definition of the plugin |  Current directory | No |
```

File: Sources/TuistDependencies/DependenciesGraph/DependenciesGraphController.swift [hunk 6]
```diff
@@ -21,7 +21,7 @@ enum DependenciesGraphControllerError: FatalError, Equatable {
     var description: String {
         switch self {
         case .failedToDecodeDependenciesGraph:
-            return "Couldn't decode the DependenciesGraph from the serialized JSON file. Running `tuist dependencies fetch` should solve the problem."
+            return "Couldn't decode the DependenciesGraph from the serialized JSON file. Running `tuist fetch dependencies` should solve the problem."
```
Comment (laxmorek) [inline Sources/TuistDependencies/DependenciesGraph/DependenciesGraphController.swift:24]: The command is `tuist dependencies fetch`. Why did you change it? 🤔

File: Sources/TuistDependencies/DependenciesGraph/DependenciesGraphController.swift [hunk 7]
```diff
@@ -80,7 +80,7 @@ public final class DependenciesGraphController: DependenciesGraphControlling {
         } catch {
             logger
                 .debug(
-                    "Failed to load dependencies graph, running `tuist dependencies fetch` should solve the problem.\nError: \(error)"
+                    "Failed to load dependencies graph, running `tuist fetch dependencies` should solve the problem.\nError: \(error)"
```
Comment (laxmorek) [inline Sources/TuistDependencies/DependenciesGraph/DependenciesGraphController.swift:83]: The command is `tuist dependencies fetch`.

File: projects/docs/docs/commands/dependencies.md [hunk 8]
```diff
@@ -6,7 +6,7 @@ description: "Learn how to use Tuist's dependencies commands to manage external
...
+External dependencies are represented by another graph. Dependency managers like [CocoaPods](https://cocoapods.org) integrate it when running `pod install` leveraging Xcode workspaces, and Swift Package Manager does it at build time leveraging Xcode's closed build system. Both approaches might lead to integration issues that can cause compilation issues down the road. We are aware that's not a great developer experience and thus we take a different approach to managing external dependencies that allows leverating Tuist features such as linting and caching. The idea is simple, developers define their Carthage and Package dependencies in a `Dependencies.swift` file. They are fetched by running `tuist fetch dependencies` and integrated into the generated Xcode project at generation time. Because we merge your project and the external dependencies' graph into a single graph, we validate and fail early if the resulting graph is invalid.
```
Comment (laxmorek) [inline projects/docs/docs/commands/dependencies.md:9]: The command is `tuist dependencies fetch`.

File: projects/docs/docs/commands/dependencies.md [hunk 9]
```diff
@@ -17,7 +17,7 @@ Check out [this page](/guides/third-party-dependencies/) for the API reference o
 Dependencies can be fetched by running the following command. They are stored in your project's `Tuist/Dependencies` directory:
 
 ```bash
-tuist dependencies fetch
+tuist fetch dependencies
```
Comment (laxmorek) [inline projects/docs/docs/commands/dependencies.md:20]: The command is `tuist dependencies fetch`.

File: Sources/TuistDependencies/DependenciesGraph/DependenciesGraphController.swift (hunk 6 above)
Comment (danyf90) [reply Sources/TuistDependencies/DependenciesGraph/DependenciesGraphController.swift:24]: It is actually changed, because now we fetch also the plugin tasks, so there is a `tuist fetch` command which does both

**Review by luispadron — APPROVED** [review]
🚀

File: Sources/TuistDependencies/DependenciesGraph/DependenciesGraphController.swift (hunk 6 above)
Comment (laxmorek) [reply Sources/TuistDependencies/DependenciesGraph/DependenciesGraphController.swift:24]: oh I haven't known, thx for your comment :)

**Review by laxmorek — APPROVED** [review]

File: projects/docs/docs/commands/plugin.md (hunk 3 above)
Comment (fortmarek) [reply projects/docs/docs/commands/plugin.md:3]: sounds good!
